src/Telegram.Bot/TelegramBotClient.cs

```diff
@@ -1573,7 +1573,28 @@ public Task<Chat> GetChatAsync(ChatId chatId, CancellationToken cancellationToke
                 {"edit_message", editMessage}
             };
 
-            return SendWebRequestAsync<Message>("setGameScore", parameters, cancellationToken);
+            object response = await SendWebRequestAsync<object>("setGameScore", parameters, cancellationToken)
+                .ConfigureAwait(false);
+
+            Message message;
+
+            if (response is Message)
+            {
+                message = response as Message;
+            }
+            else if (response is bool)
+            {
+                if ((bool)response)
+                    message = default(Message);
+                else
+                    throw new ApiRequestException("API responded with false");
+            }
+            else
+            {
+                throw new Exception("Unexpected response");
+            }
+
+            return message;
```

Comment:
can we Change this to something like this:
```csharp
switch (response)
{
    case Message message:
        return message;
    case bool b:
        if (b)
            return null;
        throw new ApiRequestException("whatever");
    default:
        throw new ApiRequestException("whatelse"); 
}

```

see: https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements